Source/ZenLib/Ztring.cpp

```diff
@@ -1312,14 +1312,15 @@ Ztring& Ztring::Date_From_Seconds_1970 (const int32s Value)
 Ztring& Ztring::Date_From_Seconds_1970 (const int64s Value)
 {
     time_t Time=(time_t)Value;
-    #if _POSIX_C_SOURCE >= 1 || _XOPEN_SOURCE || _BSD_SOURCE || _SVID_SOURCE || _POSIX_SOURCE
+    #if defined(HAVE_GMTIME_R)
     struct tm Gmt_Temp;
     struct tm *Gmt=gmtime_r(&Time, &Gmt_Temp);
     #elif defined(_MSC_VER)
     struct tm Gmt_Temp;
     errno_t gmtime_s_Result=gmtime_s(&Gmt_Temp , &Time);
     struct tm* Gmt=gmtime_s_Result?NULL:&Gmt_Temp;
     #else
+    #warning "This version of ZenLib is not thread safe"

```

Comment:
`#warning` is not in the C or C++ standard and will make [some compilers throw an error and stop compilation](https://www.keil.com/support/man/docs/armcc/armcc_chr1359124955086.htm).

Please use something like:
```
#ifdef __GNUC__
    #warning "This version of ZenLib is not thread safe"
#endif
```
(I didn't find something better, LLVM has it defined and supports it so fine... Even if I have doubts that LLVM has a platform without the functions :-p)

Reply:
Will do.